shim/shim.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+ 
+SELF=$(basename ${0})
+ACTION=${1}
+
+K8S_CLIENT=kubectl
+ROOT_DIR="$(cd "$(dirname "$0")/.." >/dev/null 2>&1 && pwd)"
+CRDS_DIR="${ROOT_DIR}/shim/crds"
+CRDS=($(ls ${CRDS_DIR}))
+
+function usage() {
+  echo -e "Usage: ${SELF} action"
+  echo -e "Actions:"
+  echo -e "  i or install:\t\tInstalls shim CRDs into a cluster"
+  echo -e "  u or uninstall:\tUninstalls shim CRDs from a cluster"
+  echo
+}
+
+function install() {
+  for CRD_FILE in "${CRDS[@]}"; do
+    CRD=${CRD_FILE%.*}
+    ${K8S_CLIENT} create -f ${CRDS_DIR}/${CRD_FILE} > /dev/null 2>&1
+    if [ $? == 0 ]; then
+      echo "customresourcedefinition.apiextensions.k8s.io/${CRD} created"
+    else 
```

Comment:
You are correct, but `kubectl` itself does not support different return values based on failure reason and I am unaware of any other way to accomplish that besides trying to parse output string which is also bad because they might change between versions.

Let's not forget this is a dev tool, not part of the product itself or production code surrounding it, and as such should we can allow some leeway here

Reply:
I was originally going to suggest parsing the output string but I didn't take into account changes between versions... but I thought of a way without parsing the output:

- Save the output string (either in-memory or in a temp file)
- If return code != 0, the print the output string (which should indicate the error)
- clear in memory buffer or delete temp file

That way we're still using the return code to check for errors and we can display the error string

Reply:
The way I see it, anything we print should be designed (as the output for a manually invoked tool is not a log) and should be well defined. Using the output of a tool used as an  implementation detail does not align with this

Reply:
Can I resolve this?

Reply:
yes